tests/components/application-connector/test/compass-runtime-agent/testkit/compassruntimeagentinit/deployment.go

```diff
@@ -0,0 +1,157 @@
+
+	err = waitForRollout(dc.deploymentName, deploymentInterface)
+	if err != nil {
+		var murliErr *multierror.Error
+
+		multierror.Append(murliErr, err)
+
+		err := rollbackDeploymentFunc()
+		if err != nil {
+			multierror.Append(murliErr, err)
+		}
+		return nil, murliErr.ErrorOrNil()
+	}
+
+	return rollbackDeploymentFunc, err
+}
+
+func newRollbackDeploymentFunc(name, previousConfigSecretNamespacedName, previousCASecretNamespacedName, previousCertSecretNamespacedName string, deploymentInterface v13.DeploymentInterface) types.RollbackFunc {
+	return func() error {
+		deployment, err := retryGetDeployment(name, deploymentInterface)
+		if err != nil {
+			return err
+		}
+
+		_, found := replaceEnvValue(deployment, ConfigurationSecretEnvName, previousConfigSecretNamespacedName)
+		if !found {
+			return fmt.Errorf("environment variable '%s' not found in %s deployment", ConfigurationSecretEnvName, name)
+		}
+
+		_, found = replaceEnvValue(deployment, CASecretEnvName, previousCASecretNamespacedName)
+		if !found {
+			return fmt.Errorf("environment variable '%s' not found in %s deployment", CASecretEnvName, name)
+		}
+
+		_, found = replaceEnvValue(deployment, ClusterCertSecretEnvName, previousCertSecretNamespacedName)
+		if !found {
+			return fmt.Errorf("environment variable '%s' not found in %s deployment", ClusterCertSecretEnvName, name)
+		}
+
+		return retryUpdateDeployment(deployment, deploymentInterface)
+	}
+}
+
+func replaceEnvValue(deployment *v12.Deployment, name, newValue string) (string, bool) {
+	envs := deployment.Spec.Template.Spec.Containers[CRAContainerNumber].Env
+	for i := range envs {
+		if envs[i].Name == name {
+			previousValue := envs[i].Value
+			envs[i].Value = newValue
+			deployment.Spec.Template.Spec.Containers[CRAContainerNumber].Env = envs
+
+			return previousValue, true
+		}
+	}
+
+	return "", false
+}
+
+func retryGetDeployment(name string, deploymentInterface v13.DeploymentInterface) (*v12.Deployment, error) {
+	var deployment *v12.Deployment
+	return deployment, retry.Do(func() error {
+		var err error
+		deployment, err = deploymentInterface.Get(context.TODO(), name, v1.GetOptions{})
+		if err != nil {
+			return err
+		}
+		return nil
+	}, retry.Attempts(RetryAttempts), retry.Delay(RetrySeconds*time.Second))
+}
+
+func retryUpdateDeployment(deployment *v12.Deployment, deploymentInterface v13.DeploymentInterface) error {
+	return retry.Do(func() error {
+		_, err := deploymentInterface.Update(context.TODO(), deployment, v1.UpdateOptions{})
+		return err
+	}, retry.Attempts(RetryAttempts), retry.Delay(RetrySeconds*time.Second))
+}
+
+func waitForRollout(name string, deploymentInterface v13.DeploymentInterface) error {
+	return retry.Do(func() error {
+		deployment, err := deploymentInterface.Get(context.TODO(), name, v1.GetOptions{})
+		if err != nil {
+			return err
+		}
+		if deployment.Status.AvailableReplicas == 0 || deployment.Status.UnavailableReplicas != 0 {
```

Comment:
We should manually check if this `if` statement works correctly and if the deployment is indeed rolled out correctly at this place

Reply:
Some initial testing shows that it is correct.